modules/storages/spec/features/storages_menu_links_spec.rb

```diff
@@ -49,14 +49,31 @@
     visit(project_path(project))
   end
 
-  it 'has links to enabled storages' do
-    visit(project_path(id: project.id))
+  context 'if user has permission to see storage links' do
+    it 'has links to enabled storages' do
+      visit(project_path(id: project.id))
+
+      expect(page).to have_link(storage.name, href: storage.host)
+      project_folder_id = project_storage_with_manual_folder.project_folder_id
+      folder_href = "#{another_storage.host}/index.php/f/#{project_folder_id}?openfile=1"
+      expect(page).to have_link(another_storage.name, href: folder_href)
+      expect(page).not_to have_link(unlinked_storage.name)
+    end
 
-    expect(page).to have_link(storage.name, href: storage.host)
-    project_folder_id = project_storage_with_manual_folder.project_folder_id
-    folder_href = "#{another_storage.host}/index.php/f/#{project_folder_id}?openfile=1"
-    expect(page).to have_link(another_storage.name, href: folder_href)
-    expect(page).not_to have_link(unlinked_storage.name)
+    context 'if user is an admin but not a member of the project' do
+      let(:user) { create(:admin) }
+
+      it 'has a link to enabled storage, but not to hidden for him project' do
+        visit(project_path(id: project.id))
+
+        expect(page).to have_link(storage.name, href: storage.host)
+        project_folder_id = project_storage_with_manual_folder.project_folder_id
+        folder_href = "#{another_storage.host}/index.php/f/#{project_folder_id}?openfile=1"
+        expect(page).not_to have_link(another_storage.name, href: folder_href)
+        expect(page).to have_link(another_storage.name, href: another_storage.host)
```

Comment:
wouldn't this check suffice? if the link has href to project folder, this expectation should fail, as it expects an href to only host url.

Reply:
It should be enough. But I would prefer to be explicit here about the absence of the project folder link.